r/configure

```diff
@@ -264,7 +264,9 @@ set_pkg_vars () {
     PKG_CFLAGS="$PKG_CFLAGS $ARROW_R_CXXFLAGS"
   fi
 
-  if [ "$UNAME" = "Darwin" ] && expr $(sw_vers -productVersion) : '10\.13'; then
+  # 'expr :' always outputs the number of matched characters, to avoid noise in the log we 
+  # redirect the output to /dev/null
```

Comment:
```suggestion
  # We use expr because the product version returns more than just 10.13 and we want to 
  # match the substring. However, expr always outputs the number of matched characters
  # to stdout, to avoid noise in the log we redirect the output to /dev/null
```